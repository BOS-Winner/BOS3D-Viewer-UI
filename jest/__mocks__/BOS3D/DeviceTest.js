const DeviceTest = {
  isTouchDevice: () => false,
};

export default DeviceTest;
