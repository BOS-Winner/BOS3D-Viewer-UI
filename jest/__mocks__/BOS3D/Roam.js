class Roam {
  constructor(props) {
    this.props = props;
  }

  getId() {
    return this.props.id;
  }

  export() {
    return this.props;
  }
}

export default Roam;
