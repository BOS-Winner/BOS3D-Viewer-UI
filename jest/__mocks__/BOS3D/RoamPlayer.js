/* eslint-disable lines-between-class-members */
class RoamPlayer {
  constructor(props) {
    this.roam = props.roamData;
  }

  addStopPlayCallback() {}
  addKeyFrameCallback() {}
  start() {}
  stop() {}
  play() {}
  continue() {}
}

export default RoamPlayer;
