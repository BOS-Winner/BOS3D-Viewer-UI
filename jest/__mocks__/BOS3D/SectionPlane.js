/* eslint-disable lines-between-class-members */
class SectionPlane {
  constructor() {
    this.coordinateSystem = {
      hide: () => {},
    };
  }

  setDirection() {}
  setPlane() {}
  hidePlane() {}
  showPlane() {}
  exit() {}
}

export default SectionPlane;
