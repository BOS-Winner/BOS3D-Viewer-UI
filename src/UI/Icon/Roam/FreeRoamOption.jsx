import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { Switch, InputNumber } from 'antd';
// import CheckBox from "Base/CheckBox";
import style from "./style.less";

/**
 * 自定义开关
 * @param {object} props 参数
 * @param {string} title 开关名称
 * @param {bool} status 开关状态
 * @param {function} onChange 开关函数
 * @returns {HTMLElements} 自定义开关
 */
function SwitchItem(props) {
  const { title, status, onChange } = props;
  return (
    <div className={style.roamSettingItem}>
      <span>{title}</span>
      <Switch checked={status} onChange={onChange} className={style.customSwitch} />
    </div>
  );
}

SwitchItem.propTypes = {
  title: PropTypes.string.isRequired,
  status: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

function FreeRoamOption(props) {
  const [hitStatus, setHitStatus] = useState(false);
  const [enableGravity, setGravity] = useState(false);
  const [manHeightVisible, showManHeight] = useState(false);
  // const [manHeight, setManHeight] = useState(props.realManHeight);
  const [immersion, setImmersion] = useState(false);
  const [viewLock, setViewLock] = useState(
    props.BOS3D.GlobalData.WalkingWithViewLock
  );

  useEffect(() => {
    const EVENT_NAME = props.BOS3D.EVENTS.ON_POINTERLOCK_EXIST;
    const fn = (ev) => {
      if (ev.type === EVENT_NAME) {
        setImmersion(false);
      }
    };
    props.viewer.registerControlEventListener(EVENT_NAME, fn);

    return () => props.viewer.unregisterControlEventListener(EVENT_NAME, fn);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  function handleManHeight(value) {
    props.onChange("manHeight", true, {
      value: Number(value),
    });
  }

  const { modelDetail } = props;
  const isPicker = false; // 是否是新数据，是就隐藏
  const settingItemList = [
    //  5.0 新数据先隐藏
    ...isPicker ? [] : [{
      title: '碰撞',
      status: hitStatus,
      onChange: () => {
        setHitStatus(!hitStatus);
        props.onChange('hit', !hitStatus);
      },
      show: true
    }],
    {
      title: '俯仰锁定',
      status: viewLock,
      onChange: () => {
        props.onChange("viewLock", !viewLock);
        setViewLock(!viewLock);
      },
      show: true
    },
    {
      title: '视角跟随',
      status: immersion,
      onChange: () => {
        props.onChange("immersion", !immersion);
        setImmersion(!immersion);
      },
      show: true
    },
    //  5.0 新数据先隐藏
    ...isPicker ? [] : [
      {
        title: '重力',
        status: enableGravity,
        onChange: () => {
          setGravity(!enableGravity);
          props.onChange("gravity", !enableGravity);
        },
        show: true
      },
      {
        title: '镜头高度',
        status: manHeightVisible,
        onChange: () => {
          showManHeight(!manHeightVisible);
        },
        show: enableGravity
      },
    ]
  ];

  return (
    <div
      role="presentation"
      onClick={(e) => e.stopPropagation()}
      className={style.roamSetting}
    >
      {
        settingItemList.map(
          item => (
            item.show ? (
              <SwitchItem
                title={item.title}
                status={item.status}
                onChange={item.onChange}
                key={item.title}
              />
            ) : null
          ))
      }
      {/* 镜头高度设置 */}
      {
        manHeightVisible && enableGravity
          ? (
            <div className={style.customNumberInputWrap}>
              <InputNumber
                size="small"
                className={style.customNumberInput}
                min={0}
                max={999}
                value={props.realManHeight}
                step={0.1}
                onChange={handleManHeight}
                stringMode
                onBlur={() => {
                  props.BOS3D.Plugins.Roam.Roam.changeRoamConfig(props.viewer, true, "manHeight", {
                    value: props.realManHeight
                  });
                }}
              />
              <span className={style.customNumberInputUnit}>m</span>
            </div>
          ) : null
      }
      {/* 无用的占位 */}
      {!manHeightVisible || !enableGravity ? (
        <div style={{ width: '84px' }} />
      ) : null}
      {!enableGravity ? (
        <div style={{ width: '84px' }} />
      ) : null}
    </div>
  );
}

FreeRoamOption.propTypes = {
  onChange: PropTypes.func.isRequired,
  realManHeight: PropTypes.number.isRequired,
  BOS3D: PropTypes.object.isRequired,
  viewer: PropTypes.object.isRequired,
  modelDetail: PropTypes.object.isRequired,
};

export default FreeRoamOption;
