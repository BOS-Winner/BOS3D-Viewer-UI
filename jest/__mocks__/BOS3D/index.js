import Viewer from './Viewer';
import DeviceTest from './DeviceTest';
import SectionPlane from './SectionPlane';
import Roam from './Roam';
import RoamPlayer from './RoamPlayer';

export default {
  Viewer,
  EVENTS: {
    ON_CLICK_PICK: 'ON_CLICK_PICK',
    ON_RECTPICK_ADD: 'ON_RECTPICK_ADD',
    ON_LOAD_COMPLETE: 'ON_LOAD_COMPLETE',
    ON_SELECTION_CHANGED: 'ON_SELECTION_CHANGED',
    ON_POINTERLOCK_EXIST: 'ON_POINTERLOCK_EXIST',
  },
  DeviceTest,
  SectionPlane,
  Roam,
  RoamPlayer,
  RotatePivotMode: {
    MOUSEPOINT: 0, SELECTION: 1, CENTER: 2, CAMERA: 3
  },
  ControlConfig: {
    NoRotate: false,
  },
  GlobalData: {
    EnableSelectionOutline: true,
    EnableSelectionBoundingBox: false,
    EnableSelectionByTranslucent: false,
    WalkingWithViewLock: false,
  },
  DrawingStyle: {
    SHADING: 0,
    BOARDLINE: 1,
    SHADINGWITHLINE: 2,
  }
};
