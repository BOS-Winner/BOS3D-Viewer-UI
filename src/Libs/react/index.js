import React from "react";
import useUpdateEffect from "./useUpdateEffect";

React.useUpdateEffect = useUpdateEffect;

export default React;
// export { default as useUpdateEffect } from "./useUpdateEffect";
