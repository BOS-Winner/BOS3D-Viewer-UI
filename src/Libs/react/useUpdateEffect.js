import { useEffect, useRef } from "react";

export default function (effect, deps) {
  const isFirstRender = useRef(true);
  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else {
      effect();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, deps);
}
