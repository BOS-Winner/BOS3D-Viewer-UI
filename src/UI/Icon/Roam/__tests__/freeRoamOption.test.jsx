import React from 'react';
import { render, act, fireEvent } from "@testing-library/react";
import BOS3D from "mock/BOS3D";
import FreeRoamOption from '../FreeRoamOption';

const onChange = jest.fn();
const viewer = new BOS3D.Viewer();
let froContainer;
let getEleByText;
let getEleByValue;

describe('freeRoamOption', () => {
  beforeEach(() => {
    const { container, getByText, getByDisplayValue } = render((
      <FreeRoamOption
        onChange={onChange}
        realManHeight={1.5}
        viewer={viewer}
        BOS3D={BOS3D}
      />
    ));
    froContainer = container;
    getEleByText = getByText;
    getEleByValue = getByDisplayValue;
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render dom', () => {
    expect(froContainer).toMatchSnapshot();
  });

  it('should change impact state', () => {
    act(() => {
      getEleByText('碰撞').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();
    expect(onChange).lastCalledWith('hit', true);
    act(() => {
      getEleByText('碰撞').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();
    expect(onChange).lastCalledWith('hit', false);
  });

  it('should change gravity state', () => {
    act(() => {
      getEleByText('重力').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();
    expect(onChange).lastCalledWith('gravity', true);
    act(() => {
      getEleByText('重力').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();
    expect(onChange).lastCalledWith('gravity', false);
  });

  it("should trigger man height dialog's state", () => {
    act(() => {
      getEleByText('重力').previousElementSibling.firstElementChild.click();
      getEleByText('镜头高度').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();

    act(() => {
      getEleByText('镜头高度').previousElementSibling.firstElementChild.click();
    });
    expect(froContainer).toMatchSnapshot();

    act(() => {
      getEleByText('镜头高度').previousElementSibling.firstElementChild.click();
      getEleByText('确定').click();
    });
    expect(froContainer).toMatchSnapshot();

    act(() => {
      getEleByText('镜头高度').previousElementSibling.firstElementChild.click();
      getEleByText('取消').click();
    });
    expect(froContainer).toMatchSnapshot();
  });

  it('should change man height', () => {
    act(() => {
      getEleByText('重力').previousElementSibling.firstElementChild.click();
      getEleByText('镜头高度').previousElementSibling.firstElementChild.click();
    });
    act(() => {
      const input = getEleByValue('1.5');
      fireEvent.change(input, {
        target: {
          value: 2.2
        }
      });
    });
    getEleByText('确定').click();
    expect(onChange).lastCalledWith('manHeight', true, { value: 2.2 });
  });
});
