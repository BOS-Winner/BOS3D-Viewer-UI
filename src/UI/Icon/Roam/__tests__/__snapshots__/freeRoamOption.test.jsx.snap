// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`freeRoamOption should change gravity state 1`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: flex;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should change gravity state 2`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: none;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should change impact state 1`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: none;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should change impact state 2`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: none;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should render dom 1`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: none;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should trigger man height dialog's state 1`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: flex;"
    >
      <div
        class="manHeight"
        style="display: flex;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should trigger man height dialog's state 2`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: flex;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should trigger man height dialog's state 3`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: flex;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;

exports[`freeRoamOption should trigger man height dialog's state 4`] = `
<div>
  <div
    class="option"
    role="presentation"
  >
    <div>
      漫游设置：
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         碰撞
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(select.png);"
        />
      </div>
      <div>
         重力
      </div>
    </div>
    <div
      class="manHeightContainer"
      style="display: flex;"
    >
      <div
        class="manHeight"
        style="display: none;"
      >
        <div>
          镜头高度：
          <input
            max="999"
            min="0.1"
            step="0.1"
            type="number"
            value="1.5"
          />
           m
        </div>
        <div
          class="freeRoamBtn"
        >
          <button
            class="cancelBtn"
            type="button"
          >
            取消
          </button>
          <button
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         镜头高度
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         视角跟随
      </div>
    </div>
    <div>
      <div
        role="presentation"
      >
        <div
          role="presentation"
          style="width: 16px; height: 16px; background-image: url(normal.png);"
        />
      </div>
      <div>
         俯仰锁定
      </div>
    </div>
  </div>
</div>
`;
